lms/djangoapps/support/static/support/jsx/entitlements/components/EntitlementForm/index.jsx

```diff
@@ -0,0 +1,163 @@
+        comments: '',
+      };
+    } else {
+      this.state = {
+        courseUuid: '',
+        mode: '',
+        username: '',
+        comments: '',
+      };
+    }
+
+    this.onClose = this.onClose.bind(this);
+    this.handleCourseUUIDChange = this.handleCourseUUIDChange.bind(this);
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handleModeChange = this.handleModeChange.bind(this);
+    this.handleCommentsChange = this.handleCommentsChange.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  onClose() {
+    this.props.closeForm();
+  }
+
+  handleCourseUUIDChange(courseUuid) {
+    this.setState({ courseUuid });
+  }
+
+  handleUsernameChange(username) {
+    this.setState({ username });
+  }
+
+  handleModeChange(mode) {
+    this.setState({ mode });
+  }
+
+  handleCommentsChange(comments) {
+    this.setState({ comments });
+  }
+
+  submitForm() {
+    const { courseUuid, username, mode, comments } = this.state;
+    const { formType, entitlement } = this.props;
+    if (formType === 'reissue') { // if there is an active entitlement we are updating an entitlement
+      this.props.reissueEntitlement({ entitlement, comments });
+    } else { // if there is no active entitlement we are creating a new entitlement
+      this.props.createEntitlement({ courseUuid, username, mode, comments });
+    }
+  }
+
+  render() {
+    const { formType, courseUuid, username, mode, comments } = this.state;
+    const isReissue = formType === 'reissue'
+    const title = isReissue ? 'Re-issue Entitlement' : 'Create Entitlement';
+
+    const body = (
+      <div>
+        <h3> {title} </h3>
+        <InputText
+          disabled={isReissue}
+          name="courseUuid"
+          label="Course UUID"
+          value={courseUuid}
+          onChange={this.handleCourseUUIDChange}
+        />
+        <InputText
+          disabled={isReissue}
+          name="username"
+          label="Username"
+          value={username}
+          onChange={this.handleUsernameChange}
+        />
+        <InputSelect
+          disabled={isReissue}
+          name="mode"
+          label="Mode"
+          value={mode}
+          options={[
+            { label: '--', value: '' },
+            { label: 'Verified', value: 'verified' },
+            { label: 'Professional', value: 'professional' },
+          ]}
+          onChange={this.handleModeChange}
+        />
+        <TextArea
+          name="comments"
+          label="Comments"
+          value={comments}
+          onChange={this.handleCommentsChange}
+        />
+        <div>
+          <Button
+            className={['btn', 'btn-secondary']}
+            label="Close"
+            onClick={this.onClose}
+          />
+          <Button
+            className={['btn', 'btn-primary']}
+            label="Submit"
+            onClick={this.submitForm}
+          />
+        </div>
+      </div>
+    );
+
+    return this.props.isOpen && body;
```

Comment:
Do you need to wrap this in a selector?

Reply:
Apparently no.

Reply:
FYI (because I just gots to know) [the React Conditional Rendering docs](https://reactjs.org/docs/conditional-rendering.html)

![image](https://user-images.githubusercontent.com/8136030/37850845-168988f2-2eb3-11e8-8d94-ad68b237296a.png)